packages/components/src/Gallery/GalleryTypes.d.ts

```diff
@@ -0,0 +1,70 @@
+import { IconNames } from "../Icon";
+
+export type Sizes = "small" | "base" | "large" | "extraLarge";
+
+export type FileIconsNames = Extract<
+  IconNames,
+  "file" | "excel" | "pdf" | "video" | "word"
+>;
+
+export interface GalleryProps {
+  /**
+   * The size of the Gallery and it's files
+   * @default "base"
+   */
+  size?: "base" | "large";
+
+  /**
+   * The files for the Gallery to display
+   */
+  files: File[];
+
+  /**
+   * The max number of thumbnails before no more thumbnails are displayed
+   * unless the user clicks an action to display the rest
+   */
+  max?: number;
+
+  /**
+   * onDelete callback - this function will be called when the delete action is triggered on a Gallery image
+   */
+  onDelete?(file: File): void;
+}
+
+export interface File {
+  /**
+   * File Identifier
+   */
+  readonly key: string;
+
+  /**
+   * The name of the file.
+   */
+  readonly name: string;
+
+  /**
+   * The [MIME](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types) type of the file
+   */
+  readonly type: string;
+
+  /**
+   * The size of the file in bytes.
+   */
+  readonly size: number;
+
+  /**
+   * The progress of a file upload between 0 and 1.
+   * - `0` represents Upload just started.
+   * - `1` represents a complete upload.
+   */
+  readonly progress: number;
+
+  /**
+   * The data url of the file.
+   */
+  src: string;
```

Comment:
`readonly`?

Also, curious as to why this is not a `Promise` while the FormatFile where it's being sent to is a `Promise`

Reply:
We don't want this being a promise, since we need to access the string directly for the Lightbox to work; I originally had this as a promise, and I tried to have the LIghtbox "dig" for the string, but I wasn't able to get that to work.

Reply:
Hmm interesting. My only concern here is that it would mean we'll have some inconsistency
- `InputFile` returns a `Promise` `src` on any file you pick
- `FormatFile` expects that the file you give it has an `src` that is a `Promise`.
- `Gallery` which uses format file (which is tied to input file) is now not using a `src` `Promise` 

🤔 Not sure if that's okay or if we really need a promise in all of the file bits in atlantis 🤔

Reply:
I see what you're saying, consistency would be nice here. However, I'm not sure we can get a promise to work with the Lightbox, this might be a trade-off we have to make to get that functionality here unfortunately 😅